### PR TITLE
Update non-radial-profile sherpa plots to support matplotlib (fix #224)

### LINKED ISSUE #224: make_instmap_weights chips deprecation
The `make_instmap_weights` script imports `pychips` via `sherpa_contrib.utils` 

https://github.com/cxcsds/ciao-contrib/blob/70fcbdc11158a719bafa0bf607336e17925b9ba4/ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/utils.py#L66

We'll need to update to use `matplotlib`

### PR DESCRIPTION
**Pull request by DougBurke**
Fix the plotting in `sherpa_contrib` - for the `utils`, `chart`, and `marx` modules - so that it uses the Sherpa plotting objects (so is backend agnostic).

 - [x] `sherpa_contrib.utils.plot_instmap_weights`
    - adds the `clearwindow` argument
    - now plots the weights after downcasting to `float32` as there was "noise" when using `float64`
      which matplotlib was displaying
 - [x] `sherpa_contrib.chart.plot_chart_spectrum`
   - adds the `overplot` and `clearwindow` arguments
   - moves the model name to the plot title and drops the dataset label
 - [x] `sherpa_contrib.marx.plot_marx_spectrum`
   - adds the `overplot` and `clearwindow` arguments
   - moves the model name to the plot title and drops the dataset label
   - plots the Y axis in units of `photon/cm^2/s/keV` to match the data needed by MARX
 - [X] update the ahelp documentation
 - [X] add the changelog entry

# Note

This does not make all Sherpa plots use matplotlib, as the `sherpa_contrib.profiles.prof_XXX` routines still need to be updated, but that's for another PR.

### CODE REVIEW

**Comment by DougBurke** [comment]
@hamogu I've updated the marx plotting code so that it is now backend agnostic (uses the sherpa choice). As part of this I've tweaked the module slightly as discussed in @c221ee7 - do these changes look okay (documentation and the change of `plot_marx_spectrum` to plot `photon/cm^2/s/keV` rather than `photon/cm^2/s`, as I wanted these routines to plot the data written out to the file [there is an argument that I'm plotting a histogram and perhaps it should be a curve to match the file contents]

File: ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py [hunk 1]
```diff
@@ -205,6 +201,16 @@ def save_chart_spectrum(outfile, clobber=True, verbose=True,
         Should a message be displayed indicating that the file
         has been created? This is only used when outfile is not
         ``None``.
+    id : int or string or None
+        If id is None then the default Sherpa id is used. This
+        dataset must have a grid and source model defined.
+    elow, ehigh, ewidth : number or None
+        Only used if all three are given, otherwise the grid
+        of the data set is used. The elo value is the start
+        of the low-energy bin, ehigh is the end of the high-energy
```
Comment (hamogu) [inline ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py]: low-energybin -> bin of lowest energy and the same for high-energy-bin. When we talk about Chandra we say things like "The low-energy signal", whereas here "low-energy" cna actually be a high energy, e.h. `elow=9, ehigh=10, ewdith=0.1`, so I suggest rewording to avoid confusion.

File: ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py [hunk 2]
```diff
@@ -228,50 +234,74 @@ def save_chart_spectrum(outfile, clobber=True, verbose=True,
...
+    Parameters
+    ----------
+    id : int or string or None
+        If id is None then the default Sherpa id is used. This
+        dataset must have a grid and source model defined.
+    elow, ehigh, ewidth : number or None
+        Only used if all three are given, otherwise the grid
+        of the data set is used. The elo value is the start
+        of the low-energy bin, ehigh is the end of the high-energy
+        bin, and ewidth the bin width, all in keV.
```
Comment (hamogu) [inline ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py]: see comment above

File: ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py [hunk 3]
```diff
@@ -228,50 +234,74 @@ def save_chart_spectrum(outfile, clobber=True, verbose=True,
...
+    hplot.xlo = vals['xlo']
+    hplot.xhi = vals['xhi']
+    hplot.y = vals['y']  # .astype(np.float32)
+    hplot.xlabel = 'Energy (keV)'
+    hplot.title = 'ChaRT Spectrum: {}'.format(vals['model'])
+
+    # LaTeX support depends on the backend
+    if backend.name == 'pylab':
+        hplot.ylabel = 'Flux (photon cm$^{-2}$ s$^{-1}$)'
+    elif backend.name == 'chips':
```
Comment (hamogu) [inline ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py]: Are we going to release this while chips is still alive?

File: ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/marx.py [hunk 4]
```diff
@@ -129,6 +127,16 @@ def save_marx_spectrum(outfile, clobber=True, verbose=True,
...
         has been created? This is only used when outfile is not
         ``None``.
+    id : int or string or None
+        If id is None then the default Sherpa id is used. This
+        dataset must have a grid and source model defined.
+    elow, ehigh, ewidth : number or None
+        Only used if all three are given, otherwise the grid
+        of the data set is used. The elo value is the start
+        of the low-energy bin, ehigh is the end of the high-energy
+        bin, and ewidth the bin width, all in keV.
```
Comment (hamogu) [inline ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/marx.py]: see comment above.

File: ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/marx.py [hunk 5]
```diff
@@ -68,7 +68,7 @@ def get_marx_spectrum(id=None, elow=None, ehigh=None, ewidth=None,
     Returns
     -------
     (energ_hi, flux_bins)
-       The energies are in keV and fluxes in photons/cm^2/s.
+       The energies are in keV and fluxes in photons/cm^2/s/keV.
```
Comment (hamogu) [inline ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/marx.py]: yes!

File: ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/marx.py [hunk 6]
```diff
@@ -139,21 +147,64 @@ def save_marx_spectrum(outfile, clobber=True, verbose=True,
...
+    Parameters
+    ----------
+    id : int or string or None
+        If id is None then the default Sherpa id is used. This
+        dataset must have a grid and source model defined.
+    elow, ehigh, ewidth : number or None
+        Only used if all three are given, otherwise the grid
+        of the data set is used. The elo value is the start
+        of the low-energy bin, ehigh is the end of the high-energy
+        bin, and ewidth the bin width, all in keV.
```
Comment (hamogu) [inline ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/marx.py]: see above.

**Comment by DougBurke** [comment]
Thanks for the comments @hamogu - I'll look at them tomorrow.

File: ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py (hunk 1 above)
Comment (DougBurke) [reply ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py]: I'm changing this to the following (in `chart.py` and `marx.py`). I am not 100% happy with using left/right edge, but I also don't think it is worth obsessing over too much.

```
    elow, ehigh, ewidth : number or None
        Only used if all three are given, otherwise the grid
        of the data set is used. The elo value gives the start
        of the grid (the left edge of the first bin) and ehi
        is the end of the grid (the right edge of the last bin),
        and ewidth the bin width, all in keV.
```

Perhaps I should centralize this documentation rather than copy-pasting, but that has a user cost (having to look it up), so I am not going to do that now.

File: ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py (hunk 3 above)
Comment (DougBurke) [reply ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py]: I think we should, and as it is a small code change to support I think it's worth it even if we don't (plus if I ever do decide to right a plotly / altair / ... backend for notebook use it reminds me to update this).

**Comment by DougBurke** [comment]
Thanks for the review @hamogu - I'm going to assume I've addressed your comments sufficiently, but please holler if not :-)

**Comment by hamogu** [comment]
Looks good to meOn Jun 7, 2019 8:54 AM, Doug Burke <notifications@github.com> wrote:Thanks for the review @hamogu - I'm going to assume I've addressed your comments sufficiently, but please holler if not :-)
